app/store/middleware.js

```diff
@@ -3,13 +3,23 @@
 
 import {Platform} from 'react-native';
 import DeviceInfo from 'react-native-device-info';
+import {batchActions} from 'redux-batched-actions';
+import AsyncStorage from '@react-native-community/async-storage';
+import {purgeStoredState} from 'redux-persist';
 
-import {ViewTypes} from 'app/constants';
+import {NavigationTypes, ViewTypes} from 'app/constants';
 import initialState from 'app/initial_state';
 import {throttle} from 'app/utils/general';
 
+import {General} from 'mattermost-redux/constants';
+import {ErrorTypes, GeneralTypes} from 'mattermost-redux/action_types';
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
+
+import {persistConfig} from 'app/store/store';
```

Comment:
this is causing a cyclic import, maybe what you can  do is export a function from this file and receive the config as a parameter and with that use it in a closure in the middleware, the function can then return an array with the applied middlewares.

let me know if you need help understanding this.

Reply:
Top catch.

Handled in 9b0c78839df58058348893a95328c4ac51de9ec9 . Please take a look.